assets/app/scripts/controllers/projects.js

```diff
@@ -24,6 +25,50 @@ angular.module('openshiftConsole')
       $('#openshift-logo').off('click.projectsPage');
     });
 
+    $scope.toggleDelete = function(project) {
+      // to toggle prompt to delete
+      if (!$scope.deleteMap.hasOwnProperty(project.metadata.name)) {
+        $scope.deleteMap[project.metadata.name] = {'deleting': false};
+      }
+      // disable the delete button if current project is being deleted
+      if (!$scope.deleteMap[project.metadata.name]['deleting']) {
+        $scope.deleteMap[project.metadata.name]['toggled'] = !$scope.deleteMap[project.metadata.name]['toggled'];
+      }
+    };
+
+    $scope.deleteProject = function(project) {
+      $scope.alerts[project.metadata.name] = {};
+      $scope.deleteMap[project.metadata.name]['toggled'] = false;
+      $scope.deleteMap[project.metadata.name]['deleting'] = true;
+      DataService.delete('projects', project.metadata.name, $scope)
+      .then(function() {
+        $scope.alerts[project.metadata.name] = {
+          type: "success",
+          message: "Project " + $filter('displayName')(project) + " was successfully deleted."
+        };
+        delete $scope.projects[project.metadata.name];
+        $scope.deleteMap[project.metadata.name]['deleting'] = false;
+        $timeout(function() {
+          delete $scope.alerts[project.metadata.name];
+        }, 10000);
+      }, function(result) {
+        var details;
+        if (result.data && result.data.message) {
+          details = result.data.message;
+        }
+        $scope.alerts[project.metadata.name] = {
+          type: "error",
+          message: "Project " + $filter('displayName')(project) + " could not be deleted.",
+          details: details
+        };
+        $scope.deleteMap[project.metadata.name]['deleting'] = false;
+        console.log(result);
+        $timeout(function() {
+          delete $scope.alerts[project.metadata.name];
+        }, 10000);
+      });
```

Comment:
refactor the common bits (setting "deleting" to false and timing out the alert removal) to a `.finally(function(){ ... })` which gets called for both success and error cases